src/bson/bson-iter.c

```diff
@@ -70,6 +70,51 @@ bson_iter_init (bson_iter_t *iter,  /* OUT */
 /*
  *--------------------------------------------------------------------------
  *
+ * bson_iter_init_from_data --
+ *
+ *       Initializes @iter to be used to iterate @data of length @length
+ *
+ * Returns:
+ *       true if bson_iter_t was initialized. otherwise false.
+ *
+ * Side effects:
+ *       @iter is initialized.
+ *
+ *--------------------------------------------------------------------------
+ */
+
+bool
+bson_iter_init_from_data (bson_iter_t *iter,   /* OUT */
+                          const uint8_t *data, /* IN */
+                          size_t length)       /* IN */
+{
+   BSON_ASSERT (iter);
+   BSON_ASSERT (data);
+
+   if (BSON_UNLIKELY ((length < 5) || (length > INT_MAX))) {
+      memset (iter, 0, sizeof *iter);
+      return false;
+   }
```

Comment:
I think we should add the same overflow checks here as we do in other bson creation functions.
e.g. making sure the bson `data` reports the same length as `length` provided